packages/server-wallet/src/models/objective.ts

```diff
@@ -134,4 +103,18 @@ export class Objective extends Model {
       .findById(objectiveId)
       .patch({status: 'succeeded'});
   }
+
+  static async forChannelIds(
+    targetChannelIds: string[],
+    tx: TransactionOrKnex
+  ): Promise<ObjectiveStoredInDB[]> {
+    const objectiveIds = (
+      await ObjectiveChannel.query(tx)
+        .column('objectiveId')
+        .select()
+        .whereIn('channelId', targetChannelIds)
+    ).map(oc => oc.objectiveId);
+
+    return (await Objective.query(tx).findByIds(objectiveIds)).map(extract);
```

Comment:
I suspect there may be a smarter / more efficient way of doing this (with a single query, say).

Reply:
It's easy when you can write SQL :P

```
SELECT objectives.*
FROM objectives
INNER JOIN objective_channels ON objectives.id = objective_channels.objective_id
WHERE channel_id in ($1)
```

Reply:
(you can use `withGraphFetched`, if your models have the correct relations, and you'll get results. But it's a bit complex.)